src/index.js

```diff
@@ -15,12 +12,35 @@ export async function extractAndOptimizeUsers(dependencyMetadata) {
   return formatResponse(author, maintainers, publishers);
 }
 
+export function extractAllAuthorsFromLibrary(library) {
+  if (!library) {
+    return [];
+  }
+
+  const authors = [];
+  for (const dep of Object.values(library.dependencies)) {
+    const { author, maintainers, publishers } = dep.metadata;
+
+    const authorsFound = formatResponse(author, maintainers, publishers);
+    if (authorsFound) {
+      for (const author of authorsFound) {
+        authors.push(author);
+      }
```

Comment:
`authors.push(...authorsFound);`

No need to execute a for iteration.

Reply:
Also if authorsFound is returned "empty" no need to protect with `if` condition.